--- v2_operations.cpp.orig	2010-12-23 01:13:40.441714590 -0800
+++ v2_operations.cpp	2010-12-23 01:37:13.320414823 -0800
@@ -58,14 +58,16 @@
 
 # else // BOOST_POSIX_API
 #   include <sys/types.h>
-#   if !defined(__APPLE__) && !defined(__OpenBSD__)
+#   if !defined(__APPLE__) && !defined(__OpenBSD__) && !defined(ANDROID)
 #     include <sys/statvfs.h>
 #     define BOOST_STATVFS statvfs
 #     define BOOST_STATVFS_F_FRSIZE vfs.f_frsize
 #   else
 #ifdef __OpenBSD__
 #     include <sys/param.h>
-#endif
+#elif defined(ANDROID) 
+#     include <sys/vfs.h> 
+#endif 
 #     include <sys/mount.h>
 #     define BOOST_STATVFS statfs
 #     define BOOST_STATVFS_F_FRSIZE static_cast<boost::uintmax_t>( vfs.f_bsize )
@@ -1262,7 +1264,11 @@
         if ( max == 0 )
         {
           errno = 0;
-          long tmp = ::pathconf( "/", _PC_NAME_MAX );
+#ifdef ANDROID
+          long tmp = 0; // is it?
+#else
+           long tmp = ::pathconf( "/", _PC_NAME_MAX );
+#endif
           if ( tmp < 0 )
           {
             if ( errno == 0 ) // indeterminate
